Merchant/App/Scanner/ScannerViewController.swift

```diff
@@ -135,9 +131,9 @@ class ScannerViewController: UIViewController, AVCaptureMetadataOutputObjectsDel
         captureSession = nil
     }
     
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        setupCamera()
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        checkPermission()
```

Comment:
@Mattijah **[QUESTION]**
Is there any reason why did you remove the setup camera here?

Reply:
Copied: _"It is not removed just moved into viewDidLoad. This method also calls setupCamera which is being called in the viewDidLoad as well, so essentially you call setupCamera twice."_